src/Guesser/TypeGuesser.php

```diff
@@ -24,50 +24,29 @@
  */
 class TypeGuesser extends AbstractTypeGuesser
 {
-    /**
-     * This is a mapping between the old deprecated value we provided in the TypeGuesser
-     * and the value we should use to get the correct SonataAdminBundle template. Making the change
-     * directly in the TypeGuesser would be a BC-break if a user overrides the templates config.
-     *
-     * NEXT_MAJOR: remove this constant.
-     *
-     * @internal
-     */
-    public const DEPRECATED_TYPES = [
-        'orm_one_to_many' => TemplateRegistry::TYPE_ONE_TO_MANY,
-        'orm_many_to_many' => TemplateRegistry::TYPE_MANY_TO_MANY,
-        'orm_many_to_one' => TemplateRegistry::TYPE_MANY_TO_ONE,
-        'orm_one_to_one' => TemplateRegistry::TYPE_ONE_TO_ONE,
-        'number' => TemplateRegistry::TYPE_FLOAT,
-    ];
-
     public function guessType(string $class, string $property, ModelManagerInterface $modelManager): ?TypeGuess
     {
         if (!$ret = $this->getParentMetadataForProperty($class, $property, $modelManager)) {
-            return new TypeGuess(TemplateRegistry::TYPE_STRING, [], Guess::LOW_CONFIDENCE);
+            return new TypeGuess(FieldDescriptionInterface::TYPE_STRING, [], Guess::LOW_CONFIDENCE);
```

Comment:
Will wait for https://github.com/sonata-project/SonataDoctrineORMAdminBundle/pull/1283 to be merged into master first